internal/client/p.go

```diff
@@ -193,6 +197,55 @@ func (pc *p) AddSubnetValidator(
 		return 0, ErrEmptyID
 	}
 
+	// make sure NodeID is a staker on the primary network
+	// TODO: official wallet client should define the error value for such case
+	// currently just returns "staking too short"
+	vs, err := pc.Client().GetCurrentValidators(constants.PrimaryNetworkID, []ids.ShortID{})
```

Comment:
Would be great to put this in a helper function like `ValidatorPeriod` that returns if a validator, its weight, and validation period

Reply:
(likewise for the subnet case)